torchgen/model.py

```diff
@@ -788,9 +796,42 @@ def __post_init__(self) -> None:
                     f"that don't have matching signatures: {test_sig} != {f.func.signature()}"
                 )
         assert self.functional.func.kind() == SchemaKind.functional
+        assert (
+            not self.functional.is_view_op
+        ), "View operator shouldn't be grouped into NativeFunctionsGroup objects. Did you add an out= variant to a view operator?"
         assert self.out.func.kind() == SchemaKind.out
+
         if self.inplace is not None:
             assert self.inplace.func.kind() == SchemaKind.inplace
+            mutable_inputs = [
+                a
+                for a in f.func.arguments.flat_all
+                if a.annotation is not None and a.annotation.is_write
+            ]
+            immutable_rets = [r for r in f.func.returns if r.annotation is None]
```

Comment:
mutable inputs has to be derived because they can be interleaved with regular arguments (ugh). But for immutable versus mutable returns, we have a convention they always have to be ordered in one way or another, right? So this separation should just be directly encoded in the model in question

Reply:
I'm always ok with extra restrictions :)